### PR TITLE
SG Board Refactor

### PR DESCRIPTION
**Pull request by shubhamgogna**
@ericthewry @bmwwoo @chuj

### CODE REVIEW

**Comment by ericthewry** [comment]
LGTM! Delete `GameGrid` if possible and :package:

File: app/services/DFS.coffee [hunk 1]
```diff
@@ -80,6 +77,14 @@ class DFS
     @solutionGrid[r1][c1] = @solutionGrid[r2][c2]
     @solutionGrid[r2][c2] = temp
 
+  @createEmptyGrid: (size) ->
```
Comment (bmwwoo) [inline app/services/DFS.coffee]: Since we have `@size`, should we not have a parameter here and just refer to that?

File: app/views/Board.coffee [hunk 2]
```diff
@@ -1,62 +1,72 @@
...
+    # Unused now, but will be used for board reset
+    @initialValues = @copyValues @boardValues
+    @dimension = @boardValues.length
     @clickHandler = new ClickHandler this, @two
 
-    @size = @two.height * .80
-    offset = @size * .025
-    width = ( @size - offset ) / ( @grid.dimension ) - offset
+    @createBoard()
+    @createEmptyCells @cellWidth - 5
```
Comment (bmwwoo) [inline app/views/Board.coffee:11]: what is the `- 5`

File: app/views/Board.coffee [hunk 3]
```diff
@@ -65,13 +75,27 @@ class Board
...
+    temp = @boardValues[r1][c1]
+    @boardValues[r1][c1] = @boardValues[r2][c2]
+    @boardValues[r2][c2] = temp
+
+  copyValues: (source) ->
+    dest = []
+    for row in [0...@dimension]
+      dest.push []
+      for col in [0...@dimension]
+        # Q: Will it copy the pointer to the char or the char?
```
Comment (bmwwoo) [inline app/views/Board.coffee]: Extra comment and I want to say it's pushing the pointer 
http://snook.ca/archives/javascript/javascript_pass

Comment (bmwwoo) [inline app/views/Board.coffee]: http://stackoverflow.com/questions/6605640/javascript-by-reference-vs-by-value
Honestly now I have no idea

**Pull request merged**